src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1510 @@ const ActorInit En_Elf_InitVars = {
+        this->actor.update = func_80A05114;
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+// ask to talk to navi
+void func_80A05208(Actor* thisx, GlobalContext* globalCtx) {
+    s32 naviCUpText;
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx))) {
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0: // yes
+                naviCUpText = ElfMessage_GetCUpText(globalCtx);
+
+                if (naviCUpText != 0) {
+                    func_8010B720(globalCtx, naviCUpText);
+                } else {
+                    func_8010B720(globalCtx, 0x15F);
+                }
+
+                this->actor.update = func_80A052F4;
+                break;
+            case 1: // no
+                func_80106CCC(globalCtx);
+                this->actor.update = func_80A053F0;
+                func_80A01C38(this, 0);
+                this->flags &= ~0x20;
+                break;
+        }
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+// ask to talk to saria
+void func_80A052F4(Actor* thisx, GlobalContext* globalCtx) {
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if (func_8010BDBC(&globalCtx->msgCtx) == 4) {
+        if (func_80106BC8(globalCtx)) {
+            globalCtx->msgCtx.unk_E3F2 = 0xFF;
+
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0: // yes
+                    this->actor.update = func_80A05188;
+                    func_8010B720(globalCtx, 0xE2);
+                    break;
+                case 1: // no
+                    this->actor.update = func_80A05208;
+                    func_8010B720(globalCtx, 0xE1);
+                    break;
+            }
+        }
+    } else if (func_8002F334(thisx, globalCtx)) {
+        this->actor.update = func_80A053F0;
+        func_80A01C38(this, 0);
+        this->flags &= ~0x20;
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+void func_80A053F0(Actor* thisx, GlobalContext* globalCtx) {
+    u8 unk2C7;
+    s32 pad;
+    Player* player = PLAYER;
+    EnElf* this = THIS;
+
+    if (player->naviMessageId == 0) {
+        if (player->unk_664 == NULL) {
+            if (((gSaveContext.naviTimer >= 600) && (gSaveContext.naviTimer <= 3000)) || (nREG(89) != 0)) {
+                player->naviMessageId = ElfMessage_GetCUpText(globalCtx);
+
+                if (player->naviMessageId == 0x15F) {
+                    player->naviMessageId = 0;
+                }
+            }
+        }
+    } else {
+        // trigger dialog instantly for negative message IDs
+        if (player->naviMessageId < 0) {
+            thisx->flags |= 0x10000;
+        }
+    }
+
+    if (func_8002F194(thisx, globalCtx)) {
+        func_800F4524(&D_801333D4, NA_SE_VO_SK_LAUGH, 0x20);
+        thisx->posRot2.pos = thisx->posRot.pos;
+
+        if (thisx->textId == ElfMessage_GetCUpText(globalCtx)) {
+            this->flags |= 0x80;
```

Comment:
`thisx->flags ` and `this->flags` being both used 10 lines apart is a bit confusing, should `this->flags` be renamed to `elfFlags` or something?

Reply:
I agree. I think we should basically never name a specific actor field the same as a common actor field for this reason.